static/app/components/smartSearchBar/index.tsx

```diff
@@ -864,6 +867,68 @@ class SmartSearchBar extends Component<Props, State> {
     return this.findTokensAtCursor(matchedTokens);
   }
 
+  /**
+   * Get the active filter
+   */
+  get cursorFilter() {
+    const matchedTokens = [Token.Filter] as const;
+    return this.findTokensAtCursor(matchedTokens);
+  }
+
+  get cursorValueIsoDate(): TokenResult<Token.ValueIso8601Date> | null {
+    const matchedTokens = [Token.ValueIso8601Date] as const;
+    return this.findTokensAtCursor(matchedTokens);
+  }
+
+  get cursorValueRelativeDate() {
+    const matchedTokens = [Token.ValueRelativeDate] as const;
+    return this.findTokensAtCursor(matchedTokens);
+  }
+
+  get currentFieldDefinition() {
+    if (!this.cursorToken || this.cursorToken.type !== Token.Filter) {
+      return null;
+    }
+
+    const tagName = getKeyName(this.cursorToken.key, {aggregateWithArgs: true});
+
+    const fieldDefinition = getFieldDefinition(tagName);
+
+    return fieldDefinition;
```

Comment:
```suggestion
    return getFieldDefinition(tagName);
```

nit: 🤷 I do like interim return variables if their names give some extra context. But in this case the name of the function is already pretty good I think.

Reply:
Agreed - not sure why I had that in there...